plugins/filter/ace_popper.py

```diff
@@ -0,0 +1,1584 @@
+    version_added: "4.4.0"
+    short_description: Remove ace entries from a acl source of truth.
+    description:
+        - This plugin removes specific keys from a provided acl data.
+        - Using the parameters below- C(acls_data | cisco.ios.ace_popper(filter_options=filter_options, match_criteria=match_criteria))
+    options:
+      data:
+        description:
+        - This option represents a list of dictionaries of acls facts.
+        - For example C(acls_data | cisco.ios.ace_popper(filter_options=filter_options, match_criteria=match_criteria)),
+          in this case C(acls_data) represents this option.
+        type: raw
+        required: True
+      filter_options:
+        description: Specify the target keys to remove in list format.
+        type: dict
+        suboptions:
+          remove:
+            description: Specify aggregate address
+            type: str
+            choices: ['first', 'all']
+            default: all
+          failed_when:
+            description: Specify aggregate mask
+            type: str
+            choices: ['missing', 'never']
+            default: missing
+          sticky:
+            description: Specify aggregate mask
+            type: bool
+            default: False
+      match_criteria:
+        description: Specify the matching configuration of target keys and data attributes.
+        type: dict
+        required: True
+        suboptions:
+          afi:
+            description: Specify afi
+            type: str
+            required: True
+          acl_name:
+            description: ACL name
+            type: str
+          source_address:
+            description: Source address of the ACE
+            type: str
+          destination_address:
+            description: Destination address of the ACE
+            type: str
+          sequence:
+            description: Sequence number of the ACE
+            type: str
+          protocol:
+            description: Protocol name
+            type: str
+          grant:
+            description: Grant type permit or deny
+            type: str
+"""
+
+EXAMPLES = r"""
+
+##Playbook
+- name: Gather ACLs config from device existing ACLs config
+  cisco.ios.ios_acls:
+    state: gathered
+  register: result_gathered
+
+- name: Setting host facts for ace_popper filter plugin
+  ansible.builtin.set_fact:
+    acls_facts: "{{ result_gathered.gathered }}"
+    filter_options:
+      sticky: true
+    match_criteria:
+      afi: "ipv4"
+      source_address: "192.0.2.0"
+      destination_address: "192.0.3.0"
+
+- name: Invoke ace_popper filter plugin
+  ansible.builtin.set_fact:
+    clean_acls: "{{ acls_facts | cisco.ios.ace_popper(filter_options=filter_options, match_criteria=match_criteria) }}"
+
+- name: Override ACLs config with device existing ACLs config
+  cisco.ios.ios_acls:
+    state: overridden
+    config: "{{ clean_acls['clean_acls']['acls'] | from_yaml }}"
+  check_mode: true
```

Comment:
If we are showing an example of overridden with check_mode to build the command set, can we complete the circle by also adding the final ios_config task that _actually_ removes the ACEs?